packages/ui/src/primitives/button.tsx

```diff
@@ -1,41 +1,85 @@
+        'focus-visible:border-gray-a8 focus-visible:ring-accent-a3',
+      ],
+    },
+    busy: {
+      false: null,
+      true: null,
+    },
+    disabled: {
+      false: null,
+      true: null,
+    },
+  },
+  compoundVariants: [
+    { busy: false, disabled: false, intent: 'primary', className: 'hover:bg-accent-10 hover:after:opacity-0' },
+    { busy: false, disabled: false, intent: 'secondary', className: 'hover:bg-gray-2' },
+    { busy: false, disabled: true, className: 'disabled:cursor-not-allowed disabled:opacity-50' },
+    { busy: true, disabled: false, className: 'cursor-wait' },
+  ],
+  defaultVariants: {
+    busy: false,
+    disabled: false,
+    intent: 'primary',
+  },
+});
+
 export const Button = React.forwardRef(function Button(
   {
     busy,
     children,
     className,
     disabled,
     icon,
+    intent,
+    type = 'button',
     ...props
-  }: React.ButtonHTMLAttributes<HTMLButtonElement> & { busy?: boolean; icon?: React.ReactNode },
+  }: React.ButtonHTMLAttributes<HTMLButtonElement> &
+    // Omit types supplied via ButtonHTMLAttributes
+    Omit<VariantProps<typeof button>, 'disabled'> & { icon?: React.ReactNode },
   forwardedRef: React.ForwardedRef<HTMLButtonElement>,
 ) {
   return (
-    // eslint-disable-next-line react/button-has-type
     <button
       ref={forwardedRef}
-      {...props}
-      className={cx(
-        'text-accent-contrast bg-accent-9 border-accent-9 focus-visible:ring-gray-a3 relative isolate inline-flex w-full select-none appearance-none items-center justify-center rounded-md border px-2 py-1.5 text-base font-medium shadow-[0_1px_1px_0_theme(colors.white/.07)_inset] outline-none ring-[0.1875rem] ring-transparent before:absolute before:inset-0 before:rounded-[calc(theme(borderRadius.md)-1px)] before:shadow-[0_1px_1px_0_theme(colors.white/.07)_inset] after:pointer-events-none after:absolute after:inset-0 after:-z-10 after:rounded-[calc(theme(borderRadius.md)-1px)] after:bg-gradient-to-b after:from-white/10 after:to-transparent',
-        busy
-          ? 'cursor-wait'
-          : disabled
-            ? 'disabled:cursor-not-allowed disabled:opacity-50'
-            : 'hover:bg-accent-10 hover:after:opacity-0',
-        className,
-      )}
+      className={button({ busy, disabled, intent, className })}
       disabled={busy || disabled}
+      // eslint-disable-next-line react/button-has-type
+      type={type}
+      {...props}
     >
       {busy ? (
-        <Spinner className='min-w-0 shrink-0 text-[1.125rem]'>Loading…</Spinner>
+        <Spinner className='shrink-0 text-[1.125rem]'>Loading…</Spinner>
       ) : (
         <>
-          <span className='min-w-0 truncate'>{children}</span>
-          {icon && <span className='ms-2 min-w-0 shrink-0 text-[0.625rem] opacity-60'>{icon}</span>}
+          <span className='truncate'>{children}</span>
+          {icon && <span className='shrink-0 text-[0.625rem] text-inherit opacity-60'>{icon}</span>}
```

Comment:
Secondary button persists the children text on busy and places the icon position at the start of the children text. Are we looking to keep parity there?

Reply:
<img width="435" alt="Screenshot 2024-07-23 at 12 49 17 PM" src="https://github.com/user-attachments/assets/a77d1516-031a-4552-b3a9-0e92053d1b46">

Reply:
Oof good spot, maybe we need an additional `hideChildrenOnBusy` prop or something – what do you think?

I can roll that out tomorrow if you approve, but equally happy if you'd rather I explore this further and allow you approve tomorrow

Reply:
Thought on naming but couldn't come up with anything better. So use your best judgement. Approving ✅